enterprise/api/v1/views.py

```diff
@@ -1483,3 +1484,63 @@ def update(self, request, *args, **kwargs):
     )
     def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
+
+    @action(methods=['post'], detail=True, url_path='link-user')
+    def link_user(self, request, pk=None):
+        """
+        Post
+        Links user using enterprise_customer_key
+        /enterprise/api/enterprise-customer-invite-key/{enterprise_customer_key}/link-user
+
+        Given a enterprise_customer_key, link user to the appropriate enterprise.
+
+        If the key is not found, returns 404
+        If the ket is not valid, return 422
+        If we create an `EnterpriseCustomerUser` return 201
+        If an `EnterpriseCustomerUser` if found return 200
+        """
+        enterprise_customer_key = get_object_or_404(
+            models.EnterpriseCustomerInviteKey,
+            uuid=pk
+        )
+
+        if not enterprise_customer_key.is_valid:
+            return Response(
+                {"detail": "Enterprise customer invite key is not valid"},
+                status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+
+        enterprise_customer = enterprise_customer_key.enterprise_customer
+
+        enterprise_user, created = models.EnterpriseCustomerUser.all_objects.get_or_create(
+            user_id=request.user.id,
+            enterprise_customer=enterprise_customer,
+        )
+
+        response_body = {
+            "enterprise_customer_slug": enterprise_customer.slug,
+            "enterprise_customer_uuid": enterprise_customer.uuid,
+        }
+        headers = self.get_success_headers(response_body)
+
+        if created:
+            enterprise_user.invite_key = enterprise_customer_key
+            enterprise_user.save()
+            track_enterprise_user_linked(
+                request.user.id,
+                pk,
+                enterprise_customer.uuid,
+            )
```

Comment:
Should `track_enterprise_user_linked` also be called below when an EnterpriseCustomerUser record already existed but we flipped `linked/active` to True?